CCSA_ChatApp.Db/Repositories/MessageRepository.cs

```diff
@@ -39,37 +39,39 @@ private async new Task  Commit()
             }
         }
 
-        public async Task CreateMessage(Message message, string senderUsername, string receiverUsername)
+        public async Task CreateMessage(Message message, Guid senderId, string receiverUsername)
         {
-            var sender = await _userRepository.GetUserByUsername(senderUsername);
+            var sender = await _userRepository.GetUserById(senderId);
             var reciever = await _userRepository.GetUserByUsername(receiverUsername);
             if (sender != null)
             {
                 if (reciever != null)
                 {
                     var messageHistory = new MessageHistory { Receiver = reciever, Sender = sender };
-                    _session.Save(message);
-                    _messageHistoryRepository.CreateMessageHistory(message, messageHistory);
+                    await _session.SaveAsync(message);
```

Comment:
You can join line 46 and 48 together to remove the nested if

Reply:
I wanted to but i considered that they are two different scenerios thats why I left it